transformations/chinese_butter_fingers_perturbation/transformation.py

```diff
@@ -0,0 +1,248 @@
+                similar_pinyins = get_characters_with_similar_pinyin(chinese_character,
+                                                                     rare_word_prob,
+                                                                     chinese_character_database,
+                                                                     common_chinese_character_database,
+                                                                     consider_tone)
+                if random.random() <= prob and similar_pinyins != "":
+
+                    new_chinese_character = random.choice(similar_pinyins)
+                else:
+                    new_chinese_character = chinese_character
+
+                butter_text += new_chinese_character
+
+        if(butter_text not in perturbed_texts):
+            perturbed_texts.append(butter_text)
+        else:
+            perturbed_texts.append("Similar output exists in perturbed outputs")
+    return perturbed_texts
+
+def get_characters_with_similar_pinyin(chinese_character,
+                                       rare_word_prob,
+                                       chinese_character_database,
+                                       common_chinese_character_database,
+                                       consider_tone):
+
+    pinyin_for_char_to_be_perturbed = ""
+    for char_dict in chinese_character_database:
+        if(chinese_character == char_dict['word']):
+            pinyin_for_char_to_be_perturbed = char_dict['pinyin']
+            break
+
+    chars_with_similar_pinyin = ""
+    if random.random() <= rare_word_prob:
+        chars_with_similar_pinyin = retrieve_from_database(chinese_character,
+                                                           chars_with_similar_pinyin,
+                                                           chinese_character_database,
+                                                           consider_tone,
+                                                           pinyin_for_char_to_be_perturbed)
+
+    else:
+        chars_with_similar_pinyin = retrieve_from_database(chinese_character,
+                                                           chars_with_similar_pinyin,
+                                                           common_chinese_character_database,
+                                                           consider_tone,
+                                                           pinyin_for_char_to_be_perturbed)
+
+    return chars_with_similar_pinyin
+
+def get_words_with_similar_pinyin(text,
+                                rare_word_prob,
+                                chinese_character_database,
+                                common_chinese_character_database,
+                                chinese_words_database,
+                                consider_tone):
+    words_and_similar_word_dict_list = []
+    for original_word in text:
+        words_and_similar_word_dict = {"perturb_word": original_word}
```

Comment:
Maybe changing this `"perturb_word"` key to `"original_word"`

Reply:
Done!